test/attribute_cache_test.rb

```diff
@@ -86,6 +86,22 @@ def test_fetching_by_attribute_delegates_to_block_if_transactions_are_open
     end
   end
 
+  def test_overriding_should_use_cache_when_fetching_by_attribute
+    IdentityCache.cache.expects(:fetch).at_least_once
```

Comment:
I don't like how `assert_queries(0)` depends `IdentityCache.fetch` being mocked out.  I think you can just warm the cache with `AssociatedRecord.fetch_name_by_id(1)` to avoid mocking